### ISSUE TITLE
can't compile after xcode upgrade

### ISSUE THREAD

**Issue by jkarneges** [Homebrew/homebrew#44369]
Compilation within the brew environment no longer works for me unless I set `HOMEBREW_SDKROOT`.

For example, I created a simple C program:
```
#include <stdio.h>

int main()
{
    printf("hello\n");
    return 0;
}
```
I can compile this just fine outside of brew. However, if I compile within brew (e.g. by running `brew sh` and then running gcc from within there), I get the error:
```
a.c:1:10: fatal error: 'stdio.h' file not found
```
Google suggests errors regarding failing to find standard headers is usually an SDK configuration issue.

I discovered that I can make compilation work within brew if I set the `HOMEBREW_SDKROOT` environment variable once inside the environment. E.g. I can run `brew sh`, then set the environment variable to /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk and then compilation will work.

So, I think the way that Homebrew is detecting the SDK must be off somewhere.. ?

**Comment by DomT4** [comment]
> So, I think the way that Homebrew is detecting the SDK must be off somewhere.. ?

We work out SDK root [here](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/os/mac.rb#L70-L81). It's essentially chained to the version of OS X you're running, which prior to Xcode 7 wasn't an issue because Xcode shipped the most-recent previous & current OS X SDKs.

For some reason, and we haven't found out why yet because Apple has been relatively mouse-like on this, Xcode 7 was released for Yosemite and El Cap (as expected) but only contained the El Cap SDK.

If there was complete compatibility between Yosemite & El Cap's SDKs we may be able to work something with that, but given we've recently had to backport a bunch of patches that previously only applied to El Cap it seems compatibility cannot be assumed. Whilst using the El Cap SDK may work in certain areas, that hasn't been consistent across everything we've seen so far.

Our workaround for now, and something `brew doctor` will nudge you to do, is to install the Command Line Tools package if you've updated to Xcode 7 on Yosemite.
